### PR TITLE
[New] `jsx-one-expression-per-line`: add `non-jsx` option to allow non-JSX children in one line

### LINKED ISSUE #3669: New option: `jsx-one-expression-per-line` `non-jsx`
Examples of **incorrect** code for this rule:
```jsx
<Hello
  foo="bar
>Children
</Hello>
```
Examples of **correct** code for this rule:
```jsx
<Hello
  foo="bar
>
  Children
</Hello>
```

### PR DESCRIPTION
**Pull request by burtek**
Fixes #3669

### CODE REVIEW

File: lib/rules/jsx-one-expression-per-line.js [hunk 1]
```diff
@@ -65,6 +65,11 @@ module.exports = {
         return;
       }
 
+      if (options.allow === 'non-jsx'
+        && !children.find((child) => (child.type === 'JSXFragment' || child.type === 'JSXElement'))) {
```
Comment (ljharb) [inline lib/rules/jsx-one-expression-per-line.js:69]:
```suggestion
      if (
        options.allow === 'non-jsx'
        && !children.find((child) => (child.type === 'JSXFragment' || child.type === 'JSXElement'))
      ) {
```

can we add some test coverage for something like `{(<div />)}` and `{(() => <div />)()}`?

**Review by ljharb — CHANGES_REQUESTED** [review]
Thanks, this looks great!

File: lib/rules/jsx-one-expression-per-line.js (hunk 1 above)
Comment (burtek) [reply lib/rules/jsx-one-expression-per-line.js:69]: Sure, once I put my kid to sleep!

Comment (ljharb) [reply lib/rules/jsx-one-expression-per-line.js:69]: No rush :-)

**Review by ljharb — APPROVED** [review]

**Pull request merged**